lib/ansible/modules/cloud/amazon/aws_s3_bucket_facts.py

```diff
@@ -0,0 +1,172 @@
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: aws_s3_bucket_facts
+short_description: Lists S3 buckets in AWS
+requirements:
+  - boto3 >= 1.4.4
+  - python >= 2.6
+description:
+    - Lists S3 buckets in AWS, supports both boto2 and boto3
+version_added: "2.4"
+author: "Gerben Geijteman (@hyperized)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Note: Only AWS S3 is currently supported
+
+# Lists all s3 buckets
+- aws_s3_bucket_facts:
+'''
+
+RETURN = '''
+Buckets:
+  description: List of buckets
+  returned: always
+  type: list
+  sample:
+    [{
+        "Name": "my_bucket",
+        "CreationDate": "2017-07-06T15:05:12+00:00"
+    }]
+Owner:
+  description: Details of bucket owner
+  returned: always
+  type: dict
+  sample:
+    {
+      "DisplayName": "my_user",
+      "ID": "d5a0152fc8eb469fcc7eed35b2da939fd9e1a9c53924b681450f5353199900f8"
+    }
+ResponseMetadata:
+  description: Details of API call
+  returned: always
+  type: dict
+  sample:
+    {
+      "HTTPHeaders": {
+        "content-type": "application/xml",
+        "date": "Thu, 27 Jul 2017 11:37:07 GMT",
+        "server": "AmazonS3",
+        "transfer-encoding": "chunked",
+        "x-amz-id-2": "LeNLtFwr0DmQD7+pb52w3A+vuR9vkVNDswfJyTHj6673yBDoYS24EShCztue+ARrxqzDGsNWmOk=",
+        "x-amz-request-id": "F47C7B7622231E44"
+      },
+      "HTTPStatusCode": 200,
+      "HostId": "LeNLtFwr0DmQD7+pb52w3A+vuR9vkVNDswfJyTHj6673yBDoYS24EShCztue+ARrxqzDGsNWmOk=",
+      "RequestId": "F47C7B7622231E44",
+      "RetryAttempts": 0
+    }
+'''
+
+import json
+import traceback
+from datetime import datetime
+from json import dumps
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec, boto3_conn, \
+    camel_dict_to_snake_dict
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+class S3ServiceManager(object):
+    """
+    Service manager for S3
+    """
+
+    def __init__(self, module):
+        """
+        Bootstrap module and connection
+        :param module:
+        """
+        self.module = module
+
+        # Set up connection
+        region, ec2_url, aws_connect_params = get_aws_connection_info(self.module, boto3=HAS_BOTO3)
+
+        if region:
+            try:
+                self.connection = boto3_conn(self.module, conn_type='client', resource='s3', region=region,
+                                             endpoint=ec2_url,
+                                             **aws_connect_params)
+            except ClientError as e:
```

Comment:
You're not using NoCredentialsError which could be thrown here (botocore.exceptions.ProfileNotFound could happen too).